src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs

```diff
@@ -242,14 +251,25 @@ public static bool IsMsBuildBasedProject(string projectFullPath)
             }
         }
 
-        private static async Task ConsumeStreamReaderAsync(StreamReader reader, StringBuilder lines)
+        private static bool IsIgnoredOutput(string line, string[] excluded)
+        {
+            return excluded.All(p => line.ToLower().Contains(p));
+        }
+
+        private static async Task ConsumeStreamReaderAsync(StreamReader reader, StringBuilder lines, string[] excluded, Func<string, string[], bool> filter)
```

Comment:
I would remove ``excluded`` from this method and just use the filter. Within the filter you can check if it needs to do any work.  I think it would be simpler this way.